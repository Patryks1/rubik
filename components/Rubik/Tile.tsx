import React, { FC } from 'react';
import { Color } from '../../interfaces';

type Props = {
  color: Color;
  index: number;
};

const Tile: FC<Props> = ({ color, index }) => (
  <div
    className='flex-[0_0_80px] h-[80px] w-[80px] border border-black'
    style={{ backgroundColor: color }}
    data-testid='tile-cell'>
    {index}
  </div>
);

export default Tile;
