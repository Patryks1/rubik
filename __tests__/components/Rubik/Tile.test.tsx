import { render, cleanup } from '@testing-library/react';
import React from 'react';
import Tile from '@/components/Rubik/Tile';

describe('Tile', () => {
  afterAll(() => {
    cleanup();
  });

  it('renders Tile', () => {
    const { container } = render(<Tile color='red' index={1} />);
    expect(container).toMatchSnapshot();
  });
});
