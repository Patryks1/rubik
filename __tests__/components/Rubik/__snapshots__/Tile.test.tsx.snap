// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders Tile 1`] = `
<div>
  <div
    class="flex-[0_0_80px] h-[80px] w-[80px] border border-black"
    data-testid="tile-cell"
    style="background-color: red;"
  >
    1
  </div>
</div>
`;
